### DISCUSSION TITLE
invalid_grant: state mismatch

### DISCUSSION THREAD

**Discussion by vmishra22** [Azure-Samples/ms-identity-python-webapp#111] [Q&A]
I have deployed the app on kubernetes and redirect url points to a url within the deployed environment, but I get the error, invalid_grant, state mismatch. 

Example:
app_url = https://backend.<url>.com/login
redirect_url = https://backend.<url>.com/auth/callback

The same code works for http://localhost

**Comment by rayluo** [comment]
Currently this sample app uses file-based session by default. If your production environment has multiple instances of this app running on different machines, the initial login attempt and the subsequent auth response may not hit the same machine, thus unable to utilize the state persisted in session. Similar issue has been reported in #99 .

You may try to use a centralized session. See hints [here](https://github.com/Azure-Samples/ms-identity-python-webapp/blob/0.7.0/app_config.py#L24-L27).
